<!doctype html>
<html>
<head>
<meta charset="utf-8">
<title>manual</title>
</head>

<body>
<h1 align='center'>Single Cell RNA Data Analysis and Visualization</h1>
<hr>
<h3>This section will introduce how to prepare input files:</h3>
<p><strong>Supported Input Formats:</strong></p>
<ol>
  <li><strong>H5 Files (Cell Ranger Output)</strong></li>
<ul>
  <li>Cell Ranger file: filtered_feature_bc_matrix.h5.</li>
  <li>Rename it to SAMPLE_NAME.h5 for proper identification. </li>
</ul>

  <li><strong>Cell Ranger Matrix Files</strong></li>
<ul>
  <li>Cell Ranger files: matrix.mtx.gz, feature.tsv.gz, barcode.tsv.gz.</li>
  <li>Rename as SAMPLE_NAME_matrix.mtx.gz, SAMPLE_NAME_features.tsv.gz, and SAMPLE_NAME_barcodes.tsv.gz, and upload together as a set.</li>
</ul>
  <li><strong>Seurat Objects</strong></li>
<ul>
<li>Format: filename.rds (Seurat object). The orig.ident attribute should match the sample name(s).</li>
</ul>
  <li><strong>Matrix count file</strong></li>
<ul>
<li>Format: Filename.txt with rows as genes and columns as sample_cellID.</li>
</ul>
</ol>
<p><strong>Data Size and Handling:</strong></p>
<ul>
  <li>The tool can handle scRNA-seq data up to 3GB in the specified formats.</li>
  <li>Supports analysis of single or multiple samples, including up to six sample groups.</li>
  <li>After data upload, users can proceed with the analysis through a step-by-step workflow for the 1st Module, with the 'Next Step' button guiding users through each tab in the process.</li>
  <li>Once the single or multiple analysis is completed, users can analysis as per their need, there is no steps involved further</li>
</ul>
<p><strong>Output and Visualizations:</strong></p>
<ul>
  <li><strong>High-Quality Plot Download:</strong> Users can download plots in seven formats: JPG, TIFF, PDF, SVG, BMP, EPS, and PS. However, a few specific plots, such as those requiring exceptionally high detail or complex rendering (e.g., network graphs or high-resolution heatmaps), are only available as PDF files to preserve their quality and detail.</li>
  <li><strong>Summary Tables:</strong> Tables are displayed using the DT package. Users can visualize up to 100 rows (default is 10) and download the entire table as a CSV file.</li>
  <li><strong>Download Seurat Object:</strong> In single or multiple sample analyses, users can download the processed results as an RDS file (Seurat Object). </li>
</ul>
<p><strong>Example Datasets:</strong></p>
<p>To ensure seamless analysis and reproducibility, ScRDAVis includes one reference dataset for each input format, sourced from NCBI, which has been pre-tested with the tool. These datasets allow users to explore the tool's functionalities and understand the analysis workflow effectively.</p>
<ul>
  <li><strong>H5 File:</strong> <a href='https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE271107'>GSE271107</a></li>
  <li><strong>Matrix Files:</strong> <a href='https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE277476'>GSE277476</a></li>
  <li><strong>Seurat Object:</strong> <a href='https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE250488'>GSE250488</a></li>
  <li><strong>Matrix count file:</strong> <a href='https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE155953'>GSE155953</a></li>
</ul>

<hr>
<h3>Step-by-Step Approach for User Interaction</h3>
<h3>1. Single or Multiple samples analysis</h3>
	<h3>1.1 Stats</h3>
<ul>
    <li><strong>Upload and Adjust Parameters:</strong>
        <ul>
            <li>Minimum cell expression per gene: Define the minimum number of cells that should express each gene.</li>
            <li>Minimum gene expression per cell: Set the minimum number of genes each cell should express.</li>
        </ul>
		
    </li>
	<li><strong>Execution:</strong>
<ul>
<li>Click the Submit button to run the analysis based on selected parameters.</li>
</ul>
</li>
    <li><strong>Output:</strong>
        <ul>
            <li>QC plots: Quality metrics before filtering <strong>(Fig. 1.1a)</strong>.</li>
            <li>Feature-Feature Relationships Plot <strong>(Fig. 1.1b)</strong> </li>
          <li>Sample(s) cell counts table <strong>(Fig. 1.1c)</strong></li>
            <img src='www/images/1.1.jpg' width='800' height='500' alt=''/>
        </ul>
    </li>
</ul>

<h3>1.2. Sample Groups and QC Filtering</h3>
<ul>
    <li><strong>Assign sample(s) group(s):</strong><br>
        
            Choose the number of group(s) based on sample grouping
		<ul>
            <li>If only one sample: Select 1 group.</li>
<li>For multiple samples in a single condition: Select 1 group.</li>
            <li>For multiple samples in different conditions: Select up to 6 groups.</li>
        </ul>
    </li>
    <li><strong>Define Filtering Parameters:</strong>
        <ul>
            <li>Set thresholds to remove low-quality cells and genes, and optionally filter out cells with high mitochondrial gene content.</li>
			 </ul>
	</li>
			<li><strong>Execution:</strong>
<ul>
	<li>Update Filtered Data: Click to apply filters and update the dataset.</li>
</ul>
</li>
            
        
    <li><strong>Outputs after Filtering:</strong>
        <ul>
            <li>QC matrics and Bar Plot <strong>(Fig. 1.2a-d)</strong></li>
            <li>Summary Table: Filtered cell count data <strong>(Fig. 1.2e,f)</strong></li>
			<img src='www/images/1.2.jpg' width='800' height='650' alt=''/>
        </ul>
    </li>
</ul>

<h3>1.3. Normalization and PCA Analysis</h3>
<ul>
    <li><strong>Normalization Methods:</strong>
        <ul>
            <li>LogNormalize: Adjusts for sequencing depth or read count differences.
                <ul>
                    <li>Additional options: Scale Factor, Variable Genes Detection methods (vst, mvp, disp), Top Variable Features to use.</li>
                </ul>
            </li>
            <li>SCT (SCTransform): Uses regularized negative binomial regression for clustering and differential expression.</li>
        </ul>
    </li>
    <li><strong>PCA Settings:</strong>
        <ul>
            <li>Choose the PCA Dimensions (typically between 1-50) for analysis.</li>
        </ul>
    </li>
	<li><strong>Execution:</strong>
<ul>
	<li>Click  Submit button to start the analysis.</li>
</ul>
</li>
    <li><strong>Outputs:</strong>
        <ul>
            <li>PCA Heatmap <strong>(Fig. 1.3a)</strong></li>
            <li>Elbow Plot <strong>(Fig. 1.3b)</strong></li>
            <li>PCA Plot (sample-wise or group-wise) <strong>(Fig. 1.3c,d)</strong></li>
			<img src='www/images/1.3.jpg' width='800' height='600' alt=''/>
        </ul>
    </li>
</ul>

<h3>1.4. Clustering</h3>
<ul>
    <li><strong>Clustering Step:</strong>
        <ul>
            <li>Find Neighbors: The users selects the dimensions to use (PCA, integrated dimensions, etc.) and k-nearest neighbors.</li>
            <li>Clustering Algorithm: Select between Louvain, SLM or Leiden algorithms for clustering.</li>
            <li>Resolution Control: The users can adjust the resolution (0.1 to 1) parameter to control the granularity of clusters.</li>
        </ul>
    </li>
    <li><strong>Dimension Reduction:</strong>
        <ul>
            <li>o Choose between UMAP or t-SNE for dimensionality reduction.
                <ul>
                    <li>For UMAP: Users can adjust parameters like min.dist, k-nearest-neighbours, and the number of dimensions.</li>
                    <li>For t-SNE: Users can adjust the number of dimensions</li>
                </ul>
            </li>
        </ul>
    </li>
    <li><strong>Integration Method:</strong>
		 <ul>
    <li>Select integration method(s): CCA, RPCA, Harmony, or JointPCA. These methods will allow users to handle dataset complexities and integrate data from multiple samples.</li>
	  </ul>
  </li>
	<li><strong>Execution:</strong>
<ul>
	<li>Click  Submit button to start the analysis.</li>
</ul>
</li>
<li><strong>Visualize and Compare:</strong>
      <ul>
          <li>Display UMAP or t-SNE plots with clustering labels and sample/condition overlays <strong>(Fig. 1.4a,c,e)</strong>.</li>
          <li>Bar charts (Fig. 4b,d,f) and tables show cell counts per cluster and per sample/condition <strong>(Fig. 1.4g-i)</strong>.</li>
		  <img src='www/images/1.4.jpg' width='800' height='700' alt=''/>
      </ul>
  </li>
</ul>

<h3>1.5. Remove Doublets</h3>
<ul>
    <li><strong>Doublet Detection:</strong> 
<ul>
    <li>Uses DoubletFinder to identify potential doublets, helping identify cells that may contain RNA from more than one original cell.</li>
    <li>Start with an estimated doublet rate of 7.5%-10%&nbsp; (0.075 to 0.1) of total cell count.</li>
	</ul>
    </li>
	<li><strong>Execution:</strong>
<ul>
	<li>Click Detect Doublet button to start the analysis.</li>
</ul>
</li>
    <li><strong>Outputs:</strong>
        <ul>
            <li>UMAP or t-SNE Plot: Shows singlet and doublet cells, color-coded by cluster, sample, or group <strong>(Fig. 1.5a-d)</strong>.</li>
            <li>Summary Table: Counts of singlet and doublet cells <strong>(Fig. 1.5e)</strong>.</li>
        </ul>
    </li>
  <li><strong>Remove or keep doublets:</strong>
	  <ul>
    <li>Users can choose to Keep or Remove Doublets. Updated plots and tables reflect the selection <strong>(Fig. 1.5f-o)</strong>.</li>
		<img src='www/images/1.5.jpg' width='800' height='700' alt=''/>
	</ul>
  </li>
</ul>

<h3>1.6. Marker Identification</h3>
<ul>
    <li><strong>Identify markers in all clusters (FindAllMarkers):</strong>
        <ul>
            <li>Customizable parameters:
				<ul>
          <li>Minimum cell percentage (min.pct) to specify the minimum fraction of cells in which a gene is expressed.</li>
          <li>Log fold-change threshold (logfc.threshold) to filter markers based on expression magnitude.</li>
          <li>Statistical test options (test.use), including Wilcoxon rank sum (wilcox), Wilcoxon-Limma hybrid (wilcox_limma), binomial (bimod), ROC, t-test, likelihood ratio test (LR), and MAST.</li>
          <li>Positive markers only (only.pos), with options for yes or no, to focus on upregulated genes in the target cluster.</li>
            <li>When using SCTransform for normalization, out tool uses PrepSCTFindMarkers preps the data for accurate differential testing by adjusting the SCT assay, making results more reliable for FindMarkers and FindAllMarkers.</li>
				</ul>
			</li>
<li>Output: Heatmap of the top 5 genes per cluster <strong>(Fig. 1.6a)</strong>, helping users visualize the distinguishing genes for each cluster and Summary table of markers or expressed genes <strong>(Fig. 1.6b)</strong>.</li>
	<img src='www/images/1.6.jpg' width='800' height='750' alt=''/>
        </ul>
    </li>
    <li><strong>Marker identification in one specific cluster or between two clusters (FindMarkers):</strong> 
  <ul>
    <li>Identifies markers for one cluster against another or against all other clusters.</li>
    <li>Includes all the customizable parameters noted above, enabling targeted cluster comparison with refined criteria.</li>
    <li>Output: A table format displaying the expressed genes for the specified clusters, ideal for in-depth comparisons.</li>
	  </ul>
</li>
    <li><strong>Conserved marker identification for one vs. all cluster or between two clusters (FindConservedMarkers):</strong>
        <ul>
            <li>Finds markers conserved across groups (e.g., conditions) for a cluster, or conserved markers between two specific clusters.</li>
          <li>Utilizes the same customizable parameters for consistency across comparisons.</li>
            <li>Output: A table format with expressed genes, providing insights into markers consistently expressed across groups or clusters.</li>
        </ul>
    </li>
</ul>

<h3>1.7. Cell Type Prediction</h3>
<ul>
  <li><strong>ScType:</strong></li>
       <ul>
         <li>Predefined Tissue Types: Users can select from 15 tissue types, including: Adrenal, Brain, Eye, Heart, Immune, Intestine, Kidney, Liver, Lung, Muscle, Pancreas, Placenta, Spleen, Stomach, Thymus.</li>
         <li>Tissue Classification: Automatically classifies the cells based on the selected tissue type.</li>
   </ul>
    <li><strong>SingleR:</strong></li>
      <ul>
         <li>Reference Datasets: Users can use reference datasets such as: Human Primary Cell Atlas, Blueprint/ENCODE, Mouse RNA-seq, Immunological Genome Project, Database of Immune Cell Expression/eQTLs/Epigenomics, Novershtern Hematopoietic data, Monaco immune data.</li>
         <li>Prediction: Predicts the cell types based on these well-known reference datasets.</li>
     </ul>
    <li><strong>GPTCelltype:</strong></li>
      <ul>
         <li>GPT Models: Utilizes various GPT models, including: GPT-4, GPT-4-turbo, GPT-4o-mini, GPT-4o, ChatGPT-4o-latest, GPT-3.5-turbo, GPT-3.5-turbo.</li>
         <li>Gene Requirements: Requires a minimum number of top genes for accurate prediction.</li>
         <li>Availability: Available via the web platform. To use it locally, users need to update their API key by setting Sys.setenv(OPENAI_API_KEY = 'your_openai_API_key') in the global.R file.</li>
      </ul>
  <li><strong>Own Cell Labels:</strong></li>
      <ul>
         <li>User-Defined Labels: Users can manually input their own cell type labels for each cluster.</li>
         <li>Cluster Grouping: If multiple clusters need the same label, users should provide the same label name for those clusters.</li>
      </ul>
      
    <li><strong>UMAP/t-SNE Labels:</strong>
		<ul>
         <li>Label display options: Users can choose to show or hide cell type labels in the UMAP or t-SNE plots.</li>
      </ul>
	</li>
	<li><strong>Execution:</strong>
<ul>
	<li>Click Detect cell type button to start the analysis.</li>
</ul>
</li>
   <li><strong>Output:</strong></li>
      <ul>
         <li>Plot: Generates an image plot showing the predicted cell types <strong>(Fig. 1.7a,b)</strong>.</li>
         <li>Summary Table: Provides a summary table with the predicted cell types and associated scores <strong>(Fig. 1.7c)</strong>.</li>
		  <img src='www/images/1.7.jpg' width='800' height='650' alt=''/>
      </ul>
</ul>

<h3>1.8. Cluster-based Plots</h3>
<ul>
    <li><strong>Gene Selection:</strong>
        <ul>
            <li>Top Genes: Users can select the top features or genes (from 2 to 10).</li>
            <li>Custom Genes: Users may also input custom gene names by selecting them from a drop-down menu (list of genes) and entering the desired gene names as a comma-separated list.</li>
        </ul>
    </li>
    <li><strong>Plot Types:</strong>
		<ul>
    <li>Multiple visualization formats are available, including: Dot Plot, Violin Plot, Ridge Plot, Feature Plot.</li>
		</ul>
  </li>
<li><strong>Grouping and Splitting:</strong>
	<ul>
	<li>Group by: Users can organize the data by Seurat clusters or labels generated from previous cell type prediction steps.</li>
	<li>Split by: If multiple samples are present, plots can be split by condition or sample to compare expression patterns across groups.</li>
	</ul>
  </li>
	<li><strong>Execution:</strong>
<ul>
	<li>Click Generate plots button to start the analysis.</li>
</ul>
</li>
    <li><strong>Output:</strong>
	<ul>
	<li>Plot: The user receives one of the chosen plot formats (violin plot, dot plot, feature plot or ridge plot) <strong>(Fig. 1.8a-d)</strong>.</li>
	<li>Summary Tables: The tool generates tables showing marker gene cell counts and cell proportions, providing an additional layer of quantitative insight <strong>(Fig. 1.8e)</strong>.</li>
	  <img src='www/images/1.8.jpg' width='800' height='600' alt=''/>
		</ul>
	</li>
</ul>

<h3>1.9. Condition-based Analysis</h3>
<ul>
    <li><strong>Group Selection:</strong> 
	<ul>
	<li>Users can compare gene expression between two conditions by selecting one group per dropdown menu.</li>
	</ul>
	</li>
    <li><strong>Customizable Parameters:</strong><ul>
	<li>Minimum Cell Percentage (min.pct): Sets the minimum fraction of cells in which a gene must be expressed.</li>
	<li>Log Fold-Change Threshold (logfc.threshold): Filters markers by expression magnitude.</li>
	<li>Statistical Tests (test.use): Users can choose from various methods, including Wilcoxon rank sum, Wilcoxon-Limma hybrid, binomial, ROC, t-test, likelihood ratio test, and MAST.</li>
	<li>Positive Markers Only (only.pos): Option to display only upregulated genes in the target cluster.</li>
	</ul>
	</li>
    <li><strong>Visualization Options:</strong>
	<ul>
	<li>Multiple formats are available, including: Dot Plot, Violin Plot, Ridge Plot, Feature Plot, Volcano Plot</li>
	</ul>
	</li>
    <li><strong>Grouping:</strong><ul>
	<li>Group By: Users can group data by Seurat clusters or predicted cell type labels.</li>
	<li>Number of Features: Allows display of a specific number of up- and down-regulated genes (e.g., 15).</li>
	</ul>
	</li>
	<li><strong>Execution:</strong>
<ul>
	<li>Click Submit button to start the analysis.</li>
</ul>
</li>
    <li><strong>Output:</strong><ul>
	<li>Plot: The users receives the chosen plot type, providing visual comparison <strong>(Fig. 1.9a-e)</strong>.</li>
	<li>Summary Tables: Table contains the differentially expressed genes between the slected groups <strong>(Fig. 1.9f)</strong>.</li>
	<li>This setup enables users to conduct detailed comparisons between conditions, facilitating insights into differential gene expression and cellular responses.</li>
	<img src='www/images/1.9.jpg' width='800' height='700' alt=''/>
    </ul>
	</li>
</ul>
	<hr>
	<h3>Subclustering</h3>
	In ScRDAVis, users can further explore specific clusters of interest by performing subclustering analysis. This feature allows for a more granular examination of cell populations within one or multiple clusters, based on the user’s selection. Similar output were generated as like as above for the selected cluster(s) or cell type(s).

<h3>2.1. Cluster Selection:</h3>
<ul>
<li>Users can choose one or multiple clusters for subclustering.</li>
<li>Clusters can be selected based on Seurat clusters or previously predicted annotation labels.</li>
</ul>
	
<h3>2.2. Subclustering Analysis Steps:</h3>
<ul>
<li>The subclustering process mirrors the main workflow, with dedicated tabs for each stage, allowing users to perform the following analyses on the selected cluster(s):
<ul>
<li><strong>Cell Stats:</strong> Overview of cell metrics within the selected clusters, including minimum gene and cell expression thresholds.</li>
<li><strong>Normalization and PCA Analysis: </strong>Options to normalize and vizualize the  PCA data for secific cluster(s). (use the same method used in the above menu)</li>
<li><strong>Clustering:</strong> Allows users to re-cluster cells within the subclusters, providing insights into finer subpopulations.</li>
<li><strong>Marker Identification:</strong> Users can identify markers specific to subclusters, with options to customize parameters for marker detection.</li>
<li><strong>Cell Type Prediction:</strong> Provides options to predict cell types within the selected subclusters using ScType, SingleR, GPTCelltype, or custom labels.</li>
<li><strong>Cluster-Based Plots:</strong> Users can visualize gene expression within subclusters through Dot, Violin, Ridge, or Feature plots.</li>
<li><strong>Condition-Based Analysis:</strong> Enables differential expression comparisons within subclusters, providing insights into condition-specific gene expression patterns.</li>
</ul>
</li>
</ul>
<hr>
<h3>3. Correlation Network Analysis</h3>
ScRDAVis includes Cluster-Based Correlation Analysis using the genesorteR package. This feature helps users explore relationships and interactions among genes within specific clusters by calculating pairwise correlations.
<ul>
<li><strong>Prerequisites:</strong>
<ul>
<li>Correlation Network Analysis becomes available after completing single or multiple samples analysis or subclustering analysis up to cell type prediction.</li>
<li>Users can choose to conduct  analysis on: Seurat clusters or predicted cell type labels from single, multiple, or subcluster analyses.&nbsp;</li>
</ul>
</li>
<li><strong>Correlation Methods:</strong>
<ul>
<li>Pearson</li>
<li>Spearman</li>
<li>Kendall</li>
</ul>
</li>
<li><strong>Execution:</strong>
<ul>
<li>Click the Cluster correlation network button to run the analysis based on selected parameters.</li>
</ul>
</li>
<li><strong>Output:</strong>
<ul>
<li>Correlation Heatmap: Displays the correlation values between genes within clusters in a matrix format <strong>(Fig. 3a)</strong>.</li>
<li>Correlation Network Plot: Depicts the relationships between genes as a network, highlighting strongly correlated pairs <strong>(Fig. 3b)</strong>.</li>
<li>Summary Table: With the complete correlation matrix for detailed analysis <strong>(Fig. 3c)</strong></li>
	<img src='www/images/3.1.jpg' width='800' height='600' alt=''/>
</ul>
</li>
</ul>
This analysis provides a deeper understanding of gene co-expression and interaction patterns within clusters, aiding in the identification of significant biological relationships.
<hr>
<h3>4. GO Term Analysis</h3>
	ScRDAVis provides integrated Gene Ontology (GO) term analysis using the clusterProfiler package, enabling users to explore biological functions, molecular mechanisms, and cellular components related to gene expression patterns in single, multiple, or subcluster analyses. Here’s how users can conduct GO analysis:.

<ul>
<li><strong>Prerequisites:</strong>
  <ul>
    <li>GO analysis becomes available after completing single or multiple samples analysis or subclustering analysis up to cell type prediction.</li>
  </ul>
</li>
<li><strong>Input Options:</strong><br>
Users can choose to conduct GO analysis on:
<ul>
<li>Seurat clusters or predicted cell type labels from single, multiple, or subcluster analyses.</li>
<li>Users can use one or multiple clusters at a time, with an adjustable parameter (p_val_adj < 0.05) for significant results.</li>
<li>A custom list of genes: Users can manually enter gene names (comma-separated) to investigate GO terms for genes of specific interest.</li>
</ul>
</li>
<li><strong>Organisms Supported for GO Term Mapping:</strong><br>
ScRDAVis supports GO analysis for five organisms, mapping gene IDs to gene symbols:
<ul>
<li>Human: org.Hs.eg.db</li>
<li>Mouse: org.Mm.eg.db</li>
<li>Rat: org.Mmu.eg.db</li>
<li>Pig: org.Ss.eg.db</li>
<li>Rhesus: org.Rn.eg.db</li>
</ul>
</li>
<li><strong>GO Term Analysis Parameters:</strong><br>
Ontology Method: Users can choose to focus on specific biological aspects or all three:
<ul>

<li>Biological Process (BP)</li>
<li>Molecular Function (MF)</li>
<li>Cellular Component (CC)</li>
<li>All: To analyze across all three categories.</li>
</ul>
</li>
<li><strong>Adjustable Parameters:</strong>
<ul>
<li>pAdjustMethod: Select the method to adjust for multiple testing.</li>
<li>pvalueCutoff: Set a cutoff for p-values.</li>
<li>qvalueCutoff: Define a q-value threshold for significance.</li>
<li>Minimum Size of Genes: Minimum number of genes required in a GO term.</li>
<li>Maximum Size of Genes: Maximum number of genes in a GO term.</li>
<li>Plot Type: Choose a visualization format (Dot Plot, Bar Plot, Net and UpSetPlot).</li>
<li>Number of Categories to Plot: Select the number of categories to display (1 to 50).</li>
</ul>
</li>
<li><strong>Execution:</strong>
<ul>
<li>Click the GO Term button to run the analysis based on selected parameters.</li>
</ul>
</li>
<li><strong>Output</strong>
<ul>
The GO Term analysis provides:
<li>Plots: Dot plot, Bar plot, UpSet plot and Network plot for the selected ontology categories <strong>(Fig. 4a-d)</strong>.</li>
<li>Summary Table: A downloadable table summarizing the GO terms, adjusted p-values, and other relevant metrics, allowing users to interpret and visualize biological insights <strong>(Fig. 4e)</strong>.</li>
	<img src='www/images/4.1.jpg' width='800' height='600' alt=''/>
</ul>
</li>
</ul>
This GO term analysis feature in ScRDAVis provides users with an accessible, visually informative, and comprehensive view of gene functionality across clusters and conditions, enabling enhanced biological interpretation of scRNA-seq data.
<hr>
<h3>5. Pathway Analysis</h3>
ScRDAVis offers pathway analysis through KEGG and Reactome databases using the clusterProfiler and ReactomePA packages. Users can gain insights into biological pathways associated with specific gene expression profiles from single, multiple, or subcluster analyses.
<ul>
<li><strong>Prerequisites:</strong>
<ul>
<li>Pathway analysis becomes available after completing single or multiple samples analysis or subclustering analysis up to cell type prediction.</li>
</ul>
</li>
<li><strong>Input Options:</strong><br>
Pathway analysis can be performed on:
<ul>
<li>Seurat clusters or predicted cell type labels from single, multiple, or subcluster analyses.</li>
<li>One or multiple clusters simultaneously, with results filtered by an adjusted p-value (p_val_adj < 0.05).</li>
<li>A custom list of genes: Users can input specific gene names (comma-separated) to focus on pathways for genes of interest.</li>
</ul>
</li>
<li><strong>Organisms Supported for Pathway Mapping:</strong><br>
ScRDAVis enables pathway mapping for multiple organisms:
<ul>
<li>KEGG Pathways: Supports human (org.Hs.eg.db), mouse (org.Mm.eg.db), and rat (org.Mmu.eg.db) for mapping gene IDs to symbols.</li>
<li>Reactome Pathways: Available for human, mouse, and rat.</li>
</ul>
</li>
<li><strong>Pathway Analysis Parameters:</strong>
<ul>
<li>pAdjustMethod: Choose a method for multiple testing correction.</li>
<li>pvalueCutoff: Set a threshold for p-values.</li>
<li>qvalueCutoff: Define a q-value cutoff for pathway significance.</li>
<li>Minimum Size of Genes: Minimum gene count per pathway.</li>
<li>Maximum Size of Genes: Maximum gene count per pathway.</li>
<li>Plot Type: Choose visualization format (Dot Plot, Bar Plot, Net and UpSetPlot Plot).</li>
<li>Number of Pathways to Plot: Select the number of pathways to display (1 to 50).</li>
</ul>
</li>
<li><strong>Execution:</strong>
<ul>
<li>Click the Pathway Analysis button to run the analysis with the selected parameters.</li>
</ul>
</li>
<li><strong>Output</strong><br>
Pathway analysis results include:
<ul>
<li>Visualizations: Dot plot, Bar plot, UpSet plot and Network plot, showcasing significant pathways <strong>(Fig. 5a-d)</strong>.</li>
<li>Summary Table: A downloadable table with pathway details, adjusted p-values, and other metrics for further exploration and interpretation <strong>(Fig. 5e)</strong>.</li>
	<img src='www/images/5.1.jpg' width='800' height='600' alt=''/>
</ul>
</li>
</ul>
The pathway analysis functionality in ScRDAVis helps users understand the biological processes and signaling pathways linked to gene expression profiles across clusters and conditions, providing a deep functional understanding of their ScRNA-seq data.
<hr>
<h3>6. GSEA Analysis</h3>
The Gene Set Enrichment Analysis (GSEA) feature in ScRDAVis leverages the fgsea package to identify enriched pathways using ranked gene lists, such as those generated from differential expression analysis. This allows users to assess pathway-level expression changes and gain insights into functional changes across clusters or conditions.
<ul>
<li><strong>Prerequisites</strong>
<ul>
<li>GSEA analysis can be conducted following single or multiple samples analysis or subclustering analysis  up to cell type prediction.</li>
</ul>
</li>
<li><strong>Input Options</strong><br>
GSEA analysis can be performed on:
<ul>
<li>Seurat clusters or predicted cell type labels derived from single, multiple, or subcluster analyses.</li>
<li>Single or multiple clusters simultaneously, with results filtered by an adjusted p-value (p_val_adj < 0.05).</li>
</ul>
</li>
<li><strong>Organisms and Gene Sets Supported</strong>
<ul>
<li>Organisms: Human and mouse gene ID mapping.</li>
<li>Gene Set Categories: Using the msigdbr package, which provides gene sets compatible with fgsea from the Molecular Signatures Database (MSigDB). Available categories include:
<ul>
<li>Hallmark Gene Sets (H)</li>
<li>Positional Gene Sets (C1)</li>
<li>Curated Gene Sets (C2)</li>
<li>Regulatory Target Gene Sets (C3)</li>
<li>Computational Gene Sets (C4)</li>
<li>Ontology Gene Sets (C5)</li>
<li>Oncogenic Signature Gene Sets (C6)</li>
<li>Immunologic Signature Gene Sets (C7)</li>
<li>Cell Type Signature Gene Sets (C8)</li>
</ul>
</li>
</ul>
</li>
<li><strong>GSEA Analysis Parameters:</strong>
<ul>
<li>scoreType: Define the scoring method for pathway enrichment.</li>
<li>Minimal Size of Genes: Minimum number of genes in a gene set.</li>
<li>Maximal Size of Genes: Maximum number of genes in a gene set.</li>
<li>Number of Permutations: Control the precision of p-value calculations.</li>
<li>Plot Type: Choose visualization format (GSEA Plot, PlotGseaTable, Bar Plot).</li>
<li>Number of Significant Pathways to Plot: Select the number of pathways to display (1 to 40).</li>
</ul>
</li>
<li><strong>Execution:</strong>
<ul>
<li>Click the GSEA Analysis button to run the analysis with the selected parameters.</li>
</ul>
</li>
<li><strong>Output</strong><br>
The GSEA analysis provides:
<ul>
<li>Visualizations:
<ul>
<li>GSEA Plot: Displays the enrichment score curve <strong>(Fig. 6a)</strong>.</li>
<li>PlotGseaTable: Shows enriched pathways and their enrichment scores <strong>(Fig. 6b)</strong>.</li>
<li>Bar Plot: Highlights top significant pathways <strong>(Fig. 6c)</strong>.</li>
</ul>
</li>
<li>Summary Table: A downloadable table of enriched pathways, adjusted p-values, and scores. If the users selects the top 10 significant pathways, the tool displays the top 5 upregulated and top 5 downregulated pathways <strong>(Fig. 6d)</strong>.</li>
	<img src='www/images/6.1.jpg' width='800' height='550' alt=''/>
</ul>
</li>
</ul>
GSEA analysis in ScRDAVis offers a powerful method for understanding pathway-level dynamics, supporting biological interpretation of ScRNA-seq data through visual and quantitative assessments of enriched pathways.
<hr>
<h3>7. Cell-Cell Communication Analysis</h3>	
ScRDAVis integrates CellChat to enable users to analyze cell-cell communication within single or multiple samples, as well as for subclusters. This analysis identifies potential ligand-receptor interactions, allowing users to explore how different cell types or clusters communicate based on gene expression patterns.
<ul>
<li><strong>Input Options</strong>
<ul>
<li>Source of Input: Users can analyze cell-cell communication using Seurat clusters or predicted cell type labels generated from single, multiple, or subcluster analysis.</li>
<li>Organisms Supported: Human and mouse datasets are available for ligand-receptor interaction mapping.</li>
</ul>
</li>
</ul>
<h3>7.1. Parameters for Cell-Cell Communication</h3>
<ul>
<li><strong>Identify Over-Expressed Genes:</strong>
<ul>
<li>Threshold of Cell Expression Percentage: Minimum percentage of cells expressing the genes.</li>
<li>Log Fold Change Threshold: Minimum log fold-change required for genes to be considered over-expressed.</li>
<li>p-Value Threshold: Statistical significance threshold.</li>
</ul>
</li>
<li><strong>Compute Communication Probability:</strong>
<ul>
<li>Expression Method: Choose how to compute the average expression per cell group (options: triMean, truncatedMean, thresholdedMean, median).</li>
</ul>
</li>
<li><strong>Filter Communication:</strong>
<ul>
<li>Minimum Cell Requirement: Minimum number of cells needed in each cell group to analyze cell-cell communication.</li>
</ul>
</li>
<li><strong>Communication Pattern Identification:</strong>
<ul>
<li>Pattern k-Value: Defines the number of communication patterns to identify.</li>
</ul>
</li>
<li><strong>Label Option:</strong>
<ul>
<li>Show or hide labels in plots.</li>
</ul>
</li>
<li><strong>Execution:</strong>
<ul>
	<li>Click to Cell-Cell communication analysis button to start the analysis.</li>
</ul>
</li>
<li><strong>Output for Cell-Cell Communication Analysis</strong><br>
The analysis generates the following visual outputs:
<ul>
<li>Interaction Plots:
<ul>
<li>Counts and Weights/Strength: Displays the frequency and intensity of interactions among cell groups <strong>(Fig. 7.1a,b)</strong>.</li>
<li>Interaction Heatmap: Shows interaction strengths across all clusters or cell types <strong>(Fig. 7.1c)</strong>.</li>
<li>Incoming and Outgoing Signaling Patterns: Visualizes communication patterns for target and secreting cells <strong>(Fig. 7.1d,e)</strong>.</li>
</ul>
</li>
<li>Interaction Table: Includes source and target cell types, ligand-receptor pairs, and interaction scores <strong>(Fig. 7.1f)</strong>.</li>
</ul>
</li>
<img src='www/images/7.1.jpg' width='800' height='700' alt=''/>
</ul>
<h3>7.2. Analyzing Specific Signaling Pathways</h3>
For a more focused analysis, users can select a specific signaling pathway from a drop-down menu, enabling detailed visualization of the chosen pathway.
<ul>
<li><strong>Outputs for Specific Signaling Pathway:</strong>
<ul>
<li>Circle Plot: Visualizes interactions among cell groups by counts <strong>(Fig. 7.2a)</strong>.</li>
<li>Chord Plot: Depicts connections between cell types via ligand-receptor pairs <strong>(Fig. 7.2b)</strong>.</li>
<li>Interaction Heatmap: Interaction strengths among clusters for the specific pathway <strong>(Fig. 7.2c)</strong>.</li>
<li>Bubble Plot and Bar Plot: Display interaction intensity for the selected pathway <strong>(Fig. 7.2d)</strong>.</li>
<li>Hierarchy Plot: Shows the hierarchical organization of cell types and their interactions <strong>(Fig. 7.2e)</strong>.</li>
<li>Bar Plot: Shows the network analysis contribution in bar plot <strong>(Fig. 7.2f)</strong>.</li>	
<li>Violin Plot: Shows expression of pathway-associated genes <strong>(Fig. 7.2g)</strong>.</li>
<li>Signaling Pathway Table: Contains source, target, ligand, receptor, and interaction details for the specific pathway <strong>(Fig. 7.2h)</strong>.</li>
<img src='www/images/7.2.jpg' width='800' height='650' alt=''/>
</ul>
</li>
</ul>
This suite of tools and visualizations enables detailed exploration of cell communication, allowing users to interpret inter-cellular signaling dynamics in ScRNA-seq datasets with biological relevance.
<hr>
<h3>8. Trajectory and Pseudotime Analysis</h3>	
ScRDAVis integrates Monocle3 for trajectory and pseudotime analysis, allowing users to study the dynamic progression of cells over pseudotime and identify genes with functional changes along this trajectory.
<ul>
<li><strong>Preparing for Trajectory and Pseudotime Analysis</strong>
<ul>
<li>Prerequisites: Users must complete analysis up to the cell type prediction step in either single or multiple sample analysis, or subclustering analysis.</li>
<li>Input Format: The tool automatically converts the Seurat object to Monocle3 format, and users can choose between Seurat clusters or predicted cell type labels as input.</li>
<li>UMAP Requirement: UMAP should be used in clustering steps for compatibility with Monocle3.</li>
</ul>
</li>
</ul>
<h3>8.1. Parameters for Learning Trajectory</h3>
<ul>
<li><strong>Partitioning Options:</strong>
<ul>
<li>use_partition: Toggle to specify partitions for different groups.</li>
<li>close_loop: Set to close or open the trajectory loop.</li>
<li>label_groups_by_cluster: Labels cell groups by cluster.</li>
<li>label_branch_points, label_roots, label_leaves: Allows labeling of key points on the trajectory (branches, roots, leaves).</li>
</ul>
</li>
<li><strong>Execution:</strong>
<ul>
<li>Once parameters are set, users can click the Learn Trajectory button to generate the trajectory plot.</li>
</ul>
</li>
<li><strong>Output:</strong>
<ul>
<li>Trajectory Plot: Displays cell progression in trajectory space, providing insight into the cellular development path <strong>(Fig. 8a)</strong>.</li>
</ul>
</li>
</ul>
<h3>8.2. Pseudotime Ordering of Cells</h3>
<ul>
<li><strong>Parameters: </strong>
  <ul>
    <li>Root Cluster Selection: Users must select one cluster to serve as the root cluster, marking the starting point of pseudotime.</li>
    <li>Labeling Options: Parameters include options to label groups by clusters, as well as marking branch points, roots, and leaves.</li>
  </ul>
</li>
	<li><strong>Execution:</strong>
<ul>
<li>Click to Submit button to start the analysis.</li>
</ul>
</li>
<li><strong>Output:</strong>
<ul>
<li>Pseudotime Plot: Cells are arranged by pseudotime, showing the developmental trajectory <strong>(Fig. 8b)</strong>.</li>
<li>Bar Chart: Cells are ordered based on both Seurat clusters and Monocle3 pseudotime <strong>(Fig. 8c)</strong>.</li>
</ul>
</li>
</ul>
<h3>8.3. Identifying Genes with Functional Changes in Pseudotime</h3>
To explore gene expression dynamics along the pseudotime trajectory, users can analyze gene expression changes:
<ul>
<li><strong>Parameters:</strong>
<ul>
<li>Neighbor Graph Selection: Users can select between Principal Graph or K-Nearest Neighbor (KNN) to model gene expression changes.</li>
</ul>
</li>
	<li><strong>Execution:</strong>
<ul>
<li>Click Find Genes Button: Begins the identification of genes whose functions vary along pseudotime.</li>
</ul>
</li>
<li><strong>Output:</strong>
<ul>
<li>Pseudotime Plot of Cells: Visual representation of cells in pseudotime with associated gene expression <strong>(Fig. 8d)</strong>.</li>
<li>Summary Table: Lists genes with dynamic functional changes along pseudotime <strong>(Fig. 8e)</strong>.</li>
</ul>
</li>
</ul>
<h3>8.4. Plotting Gene Expression in Pseudotime</h3>
Users can visualize specific genes to observe their expression patterns over pseudotime:
<ul>
<li><strong>Gene Selection: </strong>
  <ul>
    <li>Top Genes: By default, the tool plots the top 5 genes with dynamic changes, adjustable between 1 to 10 genes.</li>
    <li>Custom Genes: Users can specify a custom list of genes (comma-separated) to plot in pseudotime.</li>
	 </ul>
</li>
    <li><strong>Output: </strong>
      <ul>
        <li>Creates a feature plot to display gene expression across cells in pseudotime <strong>(Fig. 8f)</strong>.</li>
		 <img src='www/images/8.1.jpg' width='800' height='650' alt=''/>
      </ul>
  </li>
</ul>
This functionality helps users analyze and visualize gene dynamics, offering insights into cellular progression and identifying key genes in developmental pathways.
<hr>
<h3> 9. Co-Expression and TF Analysis</h3>
<h3>9.1. Co-Expression Network Analysis</h3>
ScRDAVis incorporates co-expression network analysis for ScRNA-seq data using the hdWGCNA package. This feature enables users to identify gene modules and their relationships in Seurat clusters or predicted cell type labels. 
<ul>
<li><strong>Prerequisites:</strong>
<ul>
<li>Co-expression network analysis becomes available after completing single or multiple samples analysis or subclustering analysis up to cell type prediction.</li>
<li>User can use one cluster at a time.</li>
</ul>
</li>
<li><strong>Metacell Construction:</strong><br>
Aggregates small groups of similar cells from the same biological sample. Uses the k-Nearest Neighbors (KNN) algorithm to group similar cells and compute a metacell gene expression matrix.
<ul>
<li><strong>Parameters:</strong>
<ul>
<li>k: Number of nearest neighbors for aggregation.</li>
<li>min_cells: Minimum number of cells in a group to construct metacells.</li>
<li>max_shared: Maximum number of cells shared across two metacells.</li>
<li>target_metacells: Maximum number of target metacells to construct.</li>
</ul>
</li>
</ul>
</li>
<li><strong>Co-Expression Network Construction:</strong><br>
Builds networks with customizable parameters:
<ul>
<li>softpower: Determines the scale-free topology for constructing networks.</li>
<li>networkType: Options include signed, unsigned, or signed hybrid.</li>
</ul>
</li>
<li><strong>Module Eigengenes and Connectivity:</strong>
<ul>
<li>Scales data using selectable models: linear, poisson, or negbinom.</li>
<li>Allows Harmony batch correction for harmonized module eigengenes (hMEs), selectable by the users.</li>
</ul>
</li>
<li><strong>Hub Gene Extraction:</strong>
<ul>
<li>Extracts the top N hub genes for selected modules, aiding in the identification of key regulators.</li>
</ul>
</li>
<li><strong>Execution:</strong>
<ul>
	<li>Click the WGCNA Analysis button initiates co-expression network analysis.</li>
</ul>
</li>
<li><strong>Outputs:</strong><br>
Few plots were not available in image files format so we have provided those as pdf files.
<ul>
<li>Soft Power Plots: Visualizes the selection of the optimal soft power parameter for network construction <strong>(Fig. 9.1a)</strong>.</li>
<li>Co-Expression Network Visualization: Displays modules with distinct colors representing gene clusters <strong>(Fig. 9.1b)</strong>.</li>
<li>Ranked Genes in Modules: Provides a list of genes ranked by module membership (kME) <strong>(Fig. 9.1c)</strong>.</li>
<li>Feature Plots: Highlights the expression of modules or specific genes <strong>(Fig. 9.1d)</strong>.</li>
<li>Module Relationships Plots: Correlation between modules based on harmonized module eigengenes (hMEs) <strong>(Fig. 9.1e)</strong>.</li>
<li>Seurat DotPlot with Modules: Displays module-specific gene expression across clusters <strong>(Fig. 9.1f)</strong>.</li>
<li>Individual Module Network Plots: Visualizes the gene network for specific modules <strong>(Fig. 9.1g)</strong>.</li>
<li>Module UMAP Plots: Maps modules onto UMAP visualizations for spatial context <strong>(Fig. 9.1h)</strong>.</li>
<li>Summary Table: Soft Power Table: Lists optimal soft power values <strong>(Fig. 9.1i)</strong>. Module Assignment Table: Details gene-module relationships with colors <strong>(Fig. 9.1j)</strong>. Hub Genes Table: Identifies top hub genes per module <strong>(Fig. 9.1k)</strong>.</li>
<img src='www/images/9.1.jpg' width='800' height='800' alt=''/>
</ul>
</li>
</ul>
This functionality provides a robust framework for uncovering intricate co-expression patterns and identifying key drivers in single-cell datasets.<br>
<h3>9.2. Transcription Factor Regulatory Network Analysis</h3>
Transcription Factor (TF) Regulatory Network Analysis in ScRDAVis employs the hdWGCNA package to construct and analyze  TF regulatory networks based on ScRNA-seq data. This feature allows users to identify gene modules and investigate TF-mediated regulation within clusters or predicted cell type labels.
<ul>
<li><strong>Prerequisites:</strong>
<ul>
<li>Complete single or multiple sample analysis or subclustering analysis, including cell type prediction.</li>
<li>Analysis is performed one cluster at a time.</li>
</ul>
</li>
<li><strong>TF Regulatory Network Construction:</strong>
<ul>
<li><strong>TF Binding Motif Information:</strong>
<ul>
<li>Human: EnsDb.Hsapiens.v86, BSgenome.Hsapiens.UCSC.hg38.</li>
<li>Mouse: EnsDb.Mmusculus.v79, BSgenome.Mmusculus.UCSC.mm10.</li>
<li>Motifs from the JASPAR 2020 database for multiple species.</li>
</ul>
</li>
<li><strong>Machine Learning Model:</strong>
<ul>
<li>XGBoost used to model TF regulation for each gene with:</li>
<li>max_depth : Maximum depth of a tree</li>
<li>eta : Step size shrinkage used in update to prevent overfitting</li>
<li>alpha: L1 regularization term on weights</li>
</ul>
</li>
<li><strong>TF Regulon Strategy:</strong>
<ul>
<li>Strategy A selects the top TFs for each gene by default</li>
<li>reg_thresh : Threshold for regulatory score)</li>
<li>n_tfs : The number of top TFs to keep for each gene</li>
</ul>
</li>
<li><strong>Regulon Expression Signatures:</strong>
<ul>
<li>Positive correlation: cor_thresh = 0.05. Threshold for TF-gene correlation for genes to be included in the positive regulon score</li>
<li>Negative correlation: cor_thresh = -0.05. threshold for TF-gene correlation for genes to be included in the negative regulon score</li>
</ul>
</li>
</ul>
</li>
<li><strong>Execution:</strong>
<ul>
<li>Click Transcription factor analysis button to start the analysis.</li>
</ul>
</li>	
<li><strong>Output and Visualization:</strong>
<ul>
<li>Module Regulatory Network Plots: Positive, negative, and combined regulatory network plots. Visualize TF-to-target relationships categorized by regulatory effects <strong>(Fig. 9.2.1a-d)</strong>.</li>
<li>Regulated Scores Table: Comprehensive list of TFs and their downstream targets <strong>(Fig. 9.2.1e)</strong>.</li>
	<img src='www/images/9.2.1.jpg' width='800' height='800' alt=''/>
</ul>
</li>
<li><strong>TF-Specific Visualizations:</strong><br>
Unravel regulatory mechanisms governing gene expression in cellular contexts. Identify key transcription factors and their target genes for hypothesis generation and validation. Explore positive and negative regulatory effects within gene modules.
<ul>
<li>Select a TF from a dropdown menu to generate specific plots:</li>
</ul>
</li>
<li><strong>Outputs:</strong>
<ul>
<li>UMAP Plots: Spatial distribution of the TF <strong>(Fig. 9.2.2a)</strong>.</li>
<li>Bar Plots: Contribution of the TF across modules <strong>(Fig. 9.2.2b)</strong>.</li>
<li>Network Plots: Positive, negative, and combined networks, with primary, secondary and tertiary targets <strong>(Fig. 9.2.2c-e)</strong>.</li>
<img src='www/images/9.2.2.jpg' width='800' height='800' alt=''/>
</ul>
</li>
</ul>
This functionality provides a comprehensive view of transcriptional regulation in ScRNA-seq data, enabling detailed exploration of TF-driven cellular processes.

	
<hr>
</body>
</html>
